### PR TITLE
First pass at text component using MSDF fonts

### PR DESCRIPTION
**Pull request by daredevildave**
Including simple engine example

Things to do before making public API
- set pivot point
- check support for devices without standard deviations extension ✔️ 
- anything else required?

Things to fix
- don't recreate mesh every time text changes

### CODE REVIEW

File: src/graphics/program-lib/chunks/msdf.frag [hunk 1]
```diff
@@ -0,0 +1,26 @@
+precision highp float;
+
+#extension GL_OES_standard_derivatives : enable
```
Comment (willeastcott) [inline src/graphics/program-lib/chunks/msdf.frag]: Handle the case where this isn't available.

File: examples/text/index.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+            var c = new pc.Entity();
+            c.addComponent('camera');
+            c.translate(0, 0, 10);
+
+            app.root.addChild(c);
+
+            var t = new pc.Entity();
+            t.addComponent('text', {
+                text: "Hello World. €10",
```
Comment (willeastcott) [inline examples/text/index.html]: 10 euros?

File: src/framework/components/text/system.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        this.material = null;
+        this._shader = null;
+        this._defaultTexture = new pc.Texture(app.graphicsDevice, {width:4, height:4, format:pc.PIXELFORMAT_R8_G8_B8});
+    };
+    TextComponentSystem = pc.inherits(TextComponentSystem, pc.ComponentSystem);
+
+    pc.extend(TextComponentSystem.prototype, {
+        initializeComponentData: function (component, data, properties) {
+
+            if (data.text !== undefined) component.text = data.text;
```
Comment (vkalpias) [inline src/framework/components/text/system.js]: should all those properties be set on the component or the data object?

Comment (daredevildave) [inline src/framework/components/text/system.js]: I'm using the component directly. 

Consider this an experiment in whether we should drop the `data` object altogether.

**Pull request merged**